src/lib.rs

```diff
@@ -526,13 +558,13 @@ pub(crate) fn validate_buffers<T, V: AsRef<[T]>>(
             actual: wave_in.len(),
         });
     }
-    for (chan, wave) in wave_in.iter().enumerate() {
-        let wave = wave.as_ref();
-        if wave.len() != needed_len && mask[chan] {
+    for (chan, wave_in) in wave_in.iter().enumerate().filter(|(chan, _)| mask[*chan]) {
+        let actual_len = wave_in.as_ref().len();
+        if actual_len != input_len {
```

Comment:
Should we allow the input to be longer than needed? In a way it makes sense since we allow this for the output. But then the process methods should probably also return both how many input frames it read.

Reply:
I don't think so. The input length is fixed and consumed completely. Otherwise it would be the the callers responsibility to shorten it. Only the output capacity might be longer than needed.

Reply:
This is exactly what was bothering me. Why not apply the same rules for input and output? For the resampler it's perfectly fine to get more input samples than needed. If both input and output are allowed to be longer, then one can easily resample a longer sound clip by re-slicing the input until the whole clip has been processed. Something like this:
```
fn process(indata: &[&[usize]], outdata: &[&[usize]]) -> (usize, usize) {
    println!("processing");
    return (100, 100);
}

fn main() {
    let ch1in = vec![0usize; 1024];
    let ch2in = vec![0usize; 1024];
    let mut indata = vec![&ch1in[..], &ch2in[..]];
    let ch1out = vec![0usize; 10000];
    let ch2out = vec![0usize; 10000];
    let mut outdata = vec![&ch1out[..], &ch2out[..]];
    while indata[0].len() >= 100 && outdata[0].len() >= 100 {
        let (nbr_in, nbr_out) = process(&indata, &outdata);
        for chan in indata.iter_mut() {
            *chan = &chan[nbr_in..];
        }
        for chan in outdata.iter_mut() {
            *chan = &chan[nbr_out..];
        }
    }
}
```
This is a quite common use case for this library, and this way of doing it is easier than what must be done today.

Reply:
https://github.com/HEnquist/rubato/pull/50/commits/8a89a9d691b2075f1134d5078119d995be9ffb9b